ignite/metrics/mean_average_precision.py

```diff
@@ -0,0 +1,420 @@
+from collections import defaultdict
+from typing import Any, Callable, Dict, List, Optional, Set, Tuple, Union
+
+import torch
+
+from ignite.metrics.metric import Metric, reinit__is_reduced, sync_all_reduce
+
+
+def iou(y: torch.Tensor, y_pred: torch.Tensor, crowd: List) -> torch.Tensor:
+    m = len(y_pred)
+    n = len(y)
+
+    ious = torch.zeros(m, n)
+
+    for g in range(n):
+        Y = y[g].tolist()
+        ya = Y[2] * Y[3]
+        iscrowd = crowd[g]
+        for d in range(m):
+            P = y_pred[d].tolist()
+            pa = P[2] * P[3]
+            ious[m - d - 1, g] = 0
+            w = min(P[2] + P[0], Y[2] + Y[0]) - max(P[0], Y[0])
+            h = min(P[3] + P[1], Y[3] + Y[1]) - max(P[1], Y[1])
+            if w <= 0 or h <= 0:
+                continue
+            i = w * h
+            if iscrowd:
```

Comment:
Could we have iscrowd optional ?

Reply:
We could, but since we already have crowd field in the input, better to check it here itself than checking it elsewhere first and passing None.

Reply:
I mean crowd may not be an input. Typically, this could not be an input for a lot of datasets